frontend/src/components/Form/MarkdownEditor.tsx

```diff
@@ -11,33 +18,104 @@ interface MarkdownEditorProps<T extends FieldValues> {
   className?: string;
   /** Callback function that is called when the input value changes. */
   onChange: (value: string | undefined) => void;
+  /*** The preview type of the editor. Either 'live', 'edit' or 'preview'. */
+  preview?: PreviewType;
   /** The current value of the input. */
   value: string | undefined;
+  /** List of commands that should be shown in the toolbar. */
+  commands?: ICommand[];
+  /** Whether the fullscreen toggle should be shown. */
+  showFullScreenToggle?: boolean;
+  /** Whether the preview button should be shown. */
+  enablePreview?: boolean;
+  /** Whether the split view button should be shown. */
+  enableSplitView?: boolean;
+  /** Whether the editor should be in fullscreen mode. */
+  fullScreen?: boolean;
+  /** Callback that is called when the editor enters or exits fullscreen mode. */
+  fullScreenChange?: (fullScreen: boolean) => void;
+  /** List of commands that should be shown in the toolbar when in fullscreen mode. If not set the default commands of the editor will be used. */
+  fullScreenCommands?: ICommand[];
 }
 
+export { MarkdownEditorCommands };
+
 export default function MarkdownEditor<T extends FieldValues>({
   id,
   labelText,
   className,
   onChange,
   value,
+  preview,
+  commands,
+  enablePreview = true,
+  enableSplitView = true,
+  showFullScreenToggle = true,
+  fullScreenCommands,
+  fullScreenChange,
+  fullScreen = false,
 }: MarkdownEditorProps<T>) {
   const darkMode = useDarkModeStore((state) => state.darkMode);
 
+  const editorRef = useRef<RefMDEditor>(null);
+  const [isFullScreen, setIsFullScreen] = useState(fullScreen);
+
+  // custom full screen command to set fullScreen state of the editor
+  const customFullScreenCommand: ICommand = {
+    name: DefaultFullScreenCommand.name,
+    keyCommand: DefaultFullScreenCommand.keyCommand,
+    shortcuts: DefaultFullScreenCommand.shortcuts,
+    value: DefaultFullScreenCommand.value,
+    buttonProps: DefaultFullScreenCommand.buttonProps,
+    icon: DefaultFullScreenCommand.icon,
+    execute: function execute(state, api, dispatch, executeCommandState, shortcuts) {
```

Comment:
```suggestion
    execute(state, api, dispatch, executeCommandState, shortcuts) {
```
Would make it more readable IMO